objects/src/accounts/account_id.rs

```diff
@@ -26,7 +26,7 @@ pub enum AccountType {
 ///  - 0 - full account data is stored on-chain.
 ///  - 1 - only the account hash is stored on-chain which serves as a commitment to the account state.
 /// As such the three most significant bits fully describes the type of the account.
-#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+#[derive(Debug, Default, Copy, Clone, Eq, PartialEq)]
 pub struct AccountId(Felt);
```

Comment:
I am not sure we should add `Default` derive here. Account ID of all zeros is not a valid account ID - so, we'd be allowing creating account IDs in an invalid state.

Looking through the code, I think this is used primarily for handling account code changes. So, if we remove the need to handle these from this PR - maybe it'll be simple to remove this derive as well?